### PR TITLE
fix(ssh-tunnel): fix dataset creation flow through modal for DB with tunnel

### PR DESCRIPTION
**Pull request by hughhhh**
### SUMMARY

DB's with tunnels are failing to create datasets, put in a fix to use context manager whenever checking on physical tables.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by zhaoyongjie — COMMENTED** [review]
@hughhhh I think we should try to avoid this intrusive coding style, this context function has modified too much logic, you should redesign this function.

**Comment by hughhhh** [comment]
> @hughhhh I think we should try to avoid this intrusive coding style, this context function has modified too much logic, you should redesign this function.

why do you think it's intrusive?

This pattern is the cleanest way for us to support ssh tunnel in my opinion since we have to build the tunnel within context for any request to query the db

**Review by Antonio-RiveroMartnez — APPROVED** [review]
LGTM!

File: superset/models/core.py [hunk 1]
```diff
@@ -744,13 +744,14 @@ def update_params_from_encrypted_extra(self, params: Dict[str, Any]) -> None:
...
         extra = self.get_extra()
         meta = MetaData(**extra.get("metadata_params", {}))
-        return Table(
-            table_name,
-            meta,
-            schema=schema or None,
-            autoload=True,
-            autoload_with=self._get_sqla_engine(),
-        )
+        with self.get_sqla_engine_with_context() as engine:
```
Comment (craig-rueda) [inline superset/models/core.py:747]: Is there a way to test this?

**Review by craig-rueda — APPROVED** [review]

**Comment by sadpandajoe** [comment]
🏷️ preset:2023.01

**Pull request merged**